File: ABC135/D.py
# 間に合わない…
def solve():
    MOD = 10**9+7
    S = input()
    N = len(S)
    dp = [[0 for _ in range(13)] for _ in range(N+1)]
    dp[0][0] = 1

    for i in range(N):
        if S[i] == '?':
            for j in range(10):
                for k in range(13):
                    dp[i+1][(k*10+j)%13] += dp[i][k]
        else:
            num = int(S[i])
            for j in range(13):
                dp[i+1][(j*10+num)%13] += dp[i][j]
    print(dp[N][5]%MOD)


if __name__ == '__main__':
    solve()

File: ABC135/D.nim
import sequtils, math


proc solve(): void =
    var
        MOD : int64 = 10^9+7
        S = stdin.readLine
        N = S.len
        dp : array[10^5+1, array[13, int64]]
    dp[0][0] = 1
    
    for i, ch in S:
        if ch == '?':
            for j in 0..9:
                for k in 0..12:
                    var m = (k * 10 + j) mod 13
                    dp[i+1][m] += dp[i][k]
                    dp[i+1][m] = dp[i+1][m] mod MOD
        else:
            for k in 0..12:
                var m = (k * 10 + ord(ch) - ord('0')) mod 13
                dp[i+1][m] += dp[i][k]
                dp[i+1][m] = dp[i+1][m] mod MOD
    echo dp[N][5] mod MOD
    return

solve()
